### PR TITLE
MC-18608 - Adds commitments v2 list/get

### PR DESCRIPTION
**Pull request by m-orfanos**
### Fixes [MC-18608](https://cloud-ops.atlassian.net/browse/MC-18608)

- deletes old commitments doc
- adds `/commitments` doc

### CODE REVIEW

File: source/includes/administration/_commitments.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
+## Commitments
+
+A commitment is a set of resources given to a customer at a certain price. A customer will only pay more than the negotiated price if they go over their committed resource.
+
+<!-------------------- LIST COMMITMENTS -------------------->
+### List commitments
+
+`GET /commitments?:organization_id=:id&from=2022-01-01&to=2023-01-01`
+
+Retrieves the list of discounts associated with an organization.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
Retrieves the list of commitments associated with an organization.
```

File: source/includes/administration/_commitments.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+<!-------------------- LIST COMMITMENTS -------------------->
+### List commitments
+
+`GET /commitments?:organization_id=:id&from=2022-01-01&to=2023-01-01`
+
+Retrieves the list of commitments associated with an organization.
+
+```shell
+# Retrieve organization discount list
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
# Retrieve organization commitment list
```

File: source/includes/administration/_commitments.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+   "status": "IN_PROGRESS"
+  }
+ ]
+}
+```
+
+Optional Query Parameters | &nbsp;
+--- | ---
+`organization_id`<br/>*UUID* | The id of the organization that owns the commitment. If not provided, will default to the calling user's organization.
+`from`<br/>*string* | The inclusive date in the format YYYY-MM-DD. Cannot be after `to`.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
`from`<br/>*string* | Used as part of a date range query. Denotes the inclusive start date in the format YYYY-MM-DD. Cannot be after `to`.
```

File: source/includes/administration/_commitments.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  }
+ ]
+}
+```
+
+Optional Query Parameters | &nbsp;
+--- | ---
+`organization_id`<br/>*UUID* | The id of the organization that owns the commitment. If not provided, will default to the calling user's organization.
+`from`<br/>*string* | Used as part of a date range query. Denotes the inclusive start date in the format YYYY-MM-DD. Cannot be after `to`.
+`to`<br/>*string* | The exclusive date in the format YYYY-MM-DD. Cannot be before `from`.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
`to`<br/>*string* | Used as part of a date range query. Denotes the exclusive end date in the format YYYY-MM-DD. Cannot be before `from`.
```

File: source/includes/administration/_commitments.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+`to`<br/>*string* | Used as part of a date range query. Denotes the exclusive end date in the format YYYY-MM-DD. Cannot be before `from`.
+
+The parameters `from` and `to` are used to retrieve commitments applicable during a specific date range. They must be used together otherwise the individual parameter is ignored.
+
+Attributes | &nbsp;
+---- | ---
+`id`<br/>*string* | The id of the commitment.
+`name`<br/>*string* | The name of the commitment.
+`organization`<br/>*Organization* | The organization that owns the commitment.
+`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and discounts associated to the commitment.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and negotiated price associated to the commitment.
```

File: source/includes/administration/_commitments.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+The parameters `from` and `to` are used to retrieve commitments applicable during a specific date range. They must be used together otherwise the individual parameter is ignored.
+
+Attributes | &nbsp;
+---- | ---
+`id`<br/>*string* | The id of the commitment.
+`name`<br/>*string* | The name of the commitment.
+`organization`<br/>*Organization* | The organization that owns the commitment.
+`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and negotiated price associated to the commitment.
+`pricingMethod`<br/>*enum* | The pricing method of the commitment. Possible values are: `FIXED_PRICE` and `UTILITY_DISCOUNT`. Used to define how the effective price of the commitment is calculated.
+`rateType`<br/>*enum* | The rate type of the commitment. Applies only to commitments with pricing method `UTILITY_DISCOUNT`. Possible values are: `FIXED_RATE` and `FIXED_RATE`.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
`rateType`<br/>*enum* | The rate type of the commitment. Applies only to commitments with pricing method `UTILITY_DISCOUNT`. Possible values are: `FIXED_RATE` and `VARIABLE_RATE`.
```

File: source/includes/administration/_commitments.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+`organization`<br/>*Organization* | The organization that owns the commitment.
+`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and negotiated price associated to the commitment.
+`pricingMethod`<br/>*enum* | The pricing method of the commitment. Possible values are: `FIXED_PRICE` and `UTILITY_DISCOUNT`. Used to define how the effective price of the commitment is calculated.
+`rateType`<br/>*enum* | The rate type of the commitment. Applies only to commitments with pricing method `UTILITY_DISCOUNT`. Possible values are: `FIXED_RATE` and `VARIABLE_RATE`.
+`status`<br/>*enum* | The status of the commitment. Possible values are: `IN_PROGRESS`, `UPCOMING`, and `EXPIRED`.
+`type`<br/>*enum* | The type of the commitment. Currently limited to only `RESOURCE`.
+`fixedPrice`<br/>*double* | The effective price for this commitment. Applies only to commitments with pricing method `FIXED_PRICE`.
+`startDate`<br/>*date* | The start date of the commitment.
+`endDate`<br/>*date* | The end date of the commitment.
+`terminated`<br/>*Boolean* | A flag denoting whether or not the commitment is terminated.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
`terminated`<br/>*Boolean* | A flag denoting whether or not the commitment is terminated. This value is not directly tied to the status. 
```

File: source/includes/administration/_commitments.md [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+`pricingMethod`<br/>*enum* | The pricing method of the commitment. Possible values are: `FIXED_PRICE` and `UTILITY_DISCOUNT`. Used to define how the effective price of the commitment is calculated.
+`rateType`<br/>*enum* | The rate type of the commitment. Applies only to commitments with pricing method `UTILITY_DISCOUNT`. Possible values are: `FIXED_RATE` and `FIXED_RATE`.
+`status`<br/>*enum* | The status of the commitment. Possible values are: `IN_PROGRESS`, `UPCOMING`, and `EXPIRED`.
+`type`<br/>*enum* | The type of the commitment. Currently limited to only `RESOURCE`.
+`fixedPrice`<br/>*double* | The effective price for this commitment. Applies only to commitments with pricing method `FIXED_PRICE`.
+`startDate`<br/>*date* | The start date of the commitment.
+`endDate`<br/>*date* | The end date of the commitment.
+`terminated`<br/>*Boolean* | A flag denoting whether or not the commitment is terminated.
+`createdAt`<br/>*string* | The date on which the commitment product was created.
+`updatedAt`<br/>*string* | The date on which the commitment product was updated.
```
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
`id`<br/>*string* | The id of the commitment.
`name`<br/>*string* | The name of the commitment.
`organization`<br/>*Organization* | The organization that owns the commitment.
`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and negotiated price associated to the commitment.
`pricingMethod`<br/>*enum* | The pricing method of the commitment. Possible values are: `FIXED_PRICE` and `UTILITY_DISCOUNT`. Used to define how the effective price of the commitment is calculated.
`rateType`<br/>*enum* | The rate type of the commitment. Applies only to commitments with pricing method `UTILITY_DISCOUNT`. Possible values are: `FIXED_RATE` and `VARIABLE_RATE`.
`status`<br/>*enum* | The status of the commitment. Possible values are: `IN_PROGRESS`, `UPCOMING`, and `EXPIRED`.
`type`<br/>*enum* | The type of the commitment. Currently limited to only `RESOURCE`.
`fixedPrice`<br/>*double* | The effective price for this commitment. Applies only to commitments with pricing method `FIXED_PRICE`.
`startDate`<br/>*date* | The start date of the commitment.
`endDate`<br/>*date* | The end date of the commitment.
`terminated`<br/>*Boolean* | A flag denoting whether or not the commitment is terminated. This value is not directly tied to the status. 
`createdAt`<br/>*string* | The date on which the commitment product was created.
`updatedAt`<br/>*string* | The date on which the commitment product was updated.
```

File: source/includes/administration/_commitments.md [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+`organization`<br/>*Organization* | The organization that owns the commitment.
+`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and negotiated price associated to the commitment.
+`pricingMethod`<br/>*enum* | The pricing method of the commitment. Possible values are: `FIXED_PRICE` and `UTILITY_DISCOUNT`. Used to define how the effective price of the commitment is calculated.
+`rateType`<br/>*enum* | The rate type of the commitment. Applies only to commitments with pricing method `UTILITY_DISCOUNT`. Possible values are: `FIXED_RATE` and `VARIABLE_RATE`.
+`status`<br/>*enum* | The status of the commitment. Possible values are: `IN_PROGRESS`, `UPCOMING`, and `EXPIRED`.
+`type`<br/>*enum* | The type of the commitment. Currently limited to only `RESOURCE`.
+`fixedPrice`<br/>*double* | The effective price for this commitment. Applies only to commitments with pricing method `FIXED_PRICE`.
+`startDate`<br/>*date* | The start date of the commitment.
+`endDate`<br/>*date* | The end date of the commitment.
+`terminated`<br/>*Boolean* | A flag denoting whether or not the commitment is terminated. This value is not directly tied to the status.
```
Comment (dpmurray21) [inline source/includes/administration/_commitments.md:108]: was this deprecated?

Comment (dpmurray21) [inline source/includes/administration/_commitments.md:108]: wasn't `terminated` deprecated?

File: source/includes/administration/_commitments.md [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+
+<!-------------------- GET COMMITMENT -------------------->
+### Get commitment
+
+`GET /commitments/:id`
+
+Retrieves a commitment's details.
+
+```shell
+# Retrieve organization discount list
```
Comment (dpmurray21) [inline source/includes/administration/_commitments.md]: I don't think this comment applies here

**Review by dpmurray21 — CHANGES_REQUESTED** [review]
Nice! Two quick questions

File: source/includes/administration/_commitments.md (hunk 10 above)
Comment (m-orfanos) [inline source/includes/administration/_commitments.md]:
```suggestion
# Retrieves organization's commitment
```

File: source/includes/administration/_commitments.md (hunk 9 above)
Comment (m-orfanos) [reply source/includes/administration/_commitments.md:108]: it was renamed to `deprecated` in the DB layer, but stayed the same at the resource layer

File: source/includes/administration/_commitments.md (hunk 10 above)
Comment (m-orfanos) [reply source/includes/administration/_commitments.md]: fixed, mb

**Review by dpmurray21 — APPROVED** [review]
LGTM!

**Pull request merged**